docs/tutorial/transforms.md

```diff
@@ -1,3 +1,193 @@
 # Transforms
 
-TODO
+Among arr.ai's powerful features, transform operators are perhaps the most
+heavily used. They come in four flavors, each operating on a different kind of
+input:
+
+- `->` transforms a single value.
+- `=>` transforms the members of a set.
+- `>>` transforms the entries of a sequence or dictionary.
+- `:>` transforms the attributes of a tuple.
+
+The `->` transform operator has the following syntactic structure:
+
+- `lhs -> expr` binds the name `.` to `lhs` and evaluates `expr` in this
+  context.
+- `lhs -> \pattern expr` binds `pattern` to `lhs` and evaluates `expr` in this
+  context.
+
+The other three operators follow the same syntax. However, they bind each of the
+elements of the `lhs`, not the entire `lhs`.
+
+If this isn't clear, don't worry, it'll become clear as we work through the
+examples below.
+
+## The basics
+
+Let's explore the basic usage of the above operators, one at a time.
+
+### `->`
+
+The arrow operator, `->`, can seem almost trivial:
+
+```arrai
+@> 42 -> .
+@> 42 -> . + 1
+@> "hello, " -> . ++ "world"
+```
+
+The alternative form is likewise fairly trivial:
+
+```arrai
+@> 42 -> \x x + 1
```

Comment:
Needs result. Also this can be confusing, because it seems to bind the name `x` with `42`, which wasn't explained (I thought it always binds to `.`)